client/client.ts

```diff
@@ -81,21 +81,29 @@ const socket: Socket<ServerToClientEvents, ClientToServerEvents> = io(
 
 /*
  * Iterates through entire board, asking enclave to reveal all secrets this
- * player is privy to.
- *
- * [TODO] Only ask for tiles that should be out of the fog.
+ * player is privy to. If location is given, then the update is local.
  */
-function updatePlayerView() {
-    for (let i = 0; i < BOARD_SIZE; i++) {
-        for (let j = 0; j < BOARD_SIZE; j++) {
-            const l: Location = { r: i, c: j };
-            const sig = PLAYER.genSig(Player.hForDecrypt(l));
-            socket.emit(
-                "decrypt",
-                l,
-                PLAYER.bjjPub.serialize(),
-                Utils.serializeSig(sig)
-            );
+function updatePlayerView(l?: Location) {
+    if (l) {
+        const sig = PLAYER.genSig(Player.hForDecrypt(l));
+        socket.emit(
+            "decrypt",
+            l,
+            PLAYER.bjjPub.serialize(),
+            Utils.serializeSig(sig)
+        );
+    } else {
+        for (let i = 0; i < BOARD_SIZE; i++) {
```

Comment:
what do you think of this:

instead of this, we could add another endpoint in the enclave that asks for all tiles owned by a player, validated using the player's sig. then we can remove this O(N^2) block entirely. can only update player view at a specific location. 

worth to do now bc it's how it needs to be when the game is live

Reply:
this is very useful for a player that's logging on again, but otherwise where can it be used? when a move is finalized, updateDisplay(location) should still be called, which updates one tile.

Reply:
what about this:
1. player asks for all tiles that they own
2. as a response to that request, they attempt to decrypt each one

Reply:
sounds great to me 

they attempt to decrypt each one and the surroundings**